minicriu-client.c

```diff
@@ -241,6 +245,20 @@ int minicriu_dump(void) {
 	mc_futex_restore = 1;
 	syscall(SYS_futex, &mc_futex_restore, FUTEX_WAKE, INT_MAX);
 
+	/*
+	*	Here we synchronize the threads so that we do not
+	*	munmap segments before the threads are restored
+	*/
+
+	while ((current_count = mc_futex_checkpoint) != thread_n) {
+		syscall(SYS_futex, &mc_futex_checkpoint, FUTEX_WAIT, current_count);
+	}
+
+	maps *newmaps = getmap();
```

Comment:
Why do you need `newmaps` at all? Is not it enough to go through old mmaps and unmmap holes between two items?